homeassistant/components/template/lock.py

```diff
@@ -198,6 +204,20 @@ async def async_unlock(self, **kwargs: Any) -> None:
             self._command_unlock, run_variables=tpl_vars, context=self._context
         )
 
+    async def async_open(self, **kwargs: Any) -> None:
+        """Open the device."""
+        self._raise_template_error_if_available()
+
+        if self._optimistic:
+            self._state = False
```

Comment:
This sets the state to unlocked and not to open.

```suggestion
            self._state = STATE_OPEN
```

Reply:
changed